lib/blockchain/chain.js

```diff
@@ -1891,7 +1898,12 @@ class Chain extends AsyncEmitter {
         entry.height, util.hex32(entry.version));
     }
 
-    await this.db.save(entry, block);
+    try {
+      await this.db.save(entry, block);
```

Comment:
Maybe add modifications to the `chaindb.commit`:
```diff
diff --git a/lib/blockchain/chaindb.js b/lib/blockchain/chaindb.js
index 4b3dfcdc..b09574e0 100644
--- a/lib/blockchain/chaindb.js
+++ b/lib/blockchain/chaindb.js
@@ -295,11 +295,24 @@ class ChainDB {
   }
 
   /**
-   * Commit current batch.
+   * try committing current batch.
    * @returns {Promise}
    */
 
   async commit() {
+    try {
+      return await this._commit();
+    } catch (e) {
+      throw new CriticalError(e.message);
+    }
+  }
+
+  /**
+   * Commit current batch.
+   * @returns {Promise}
+   */
+
+  async _commit() {
     assert(this.current);
     assert(this.pending);
 
```

Reply:
Yeah this makes sense. I didn't add anything to chaindb because it doesnt extend EventEmitter. But using a new error class is the right way.